### PR TITLE
[WIP] Add standalone batch norm support via depthwise conv conversion.

### LINKED ISSUE #11586: Support Standalone Batch Norm
### 🚀 The feature, motivation and pitch

Currently XNNPACK doesn't support Batch Norm, however, we can generally support it by fusing it with convolution before hand. However for standalone batch norms we can also convert them into depthwise conv1ds. Take a look at the following code here:
https://github.com/pytorch/executorch/blob/main/backends/xnnpack/_passes/fuse_batch_norm_with_conv.py

### Alternatives

_No response_

### Additional context

_No response_

### RFC (Optional)

_No response_

### PR DESCRIPTION
**Pull request by ayushrajdahal**
### Summary

Implement ConvertBatchNormToDepthwiseConvPass to handle standalone batch normalization operations that cannot be fused with preceding convolutions.

Fixes #11586

**Key changes:**
- New pass converts standalone BatchNorm to equivalent 1x1 depthwise convolutions
- Updated BatchNormConfig to support both fusable and standalone batch norms  
- Added pass to XNNPACKPassManager before existing fusion pass
- Updated tests to expect successful partitioning of standalone batch norms

**Uses mathematical equivalence:**
  conv_weight = bn_weight / sqrt(bn_var + eps)
  conv_bias = bn_bias - bn_mean * conv_weight

### Test plan
```python
# test fused batch norm
python -m pytest backends/xnnpack/test/passes/test_batch_norm_fusion.py -v

# for standalone batch norm
python -m pytest backends/xnnpack/test/passes/test_batch_norm_to_depthwise_conv.py -v
```

### CODE REVIEW

**Comment by keyprocedure** [comment]
Hi @penknife6153, I worked on the [linear fused batchnorm PR](https://github.com/pytorch/executorch/pull/11805) and noticed some overlap between our work. Let me know if I can help with anything!
